In [1]:
import numpy as np
import pandas as pd

from decimal import *

In [2]:
data = pd.read_excel("2024 training data 3.xlsx")
teams = pd.read_excel("final 2024.xlsx")["Team"]
teams

0                Akron
1              Alabama
2              Arizona
3               Auburn
4               Baylor
            ...       
63              Wagner
64    Washington State
65    Western Kentucky
66           Wisconsin
67                Yale
Name: Team, Length: 68, dtype: object

In [3]:
final = pd.DataFrame()
final["Team"] = teams

w = []
wl = []
sos = []

for i in teams:
    print(i)
    w.append(int(list(data[data["School"] == i + " NCAA"]["W"])[0]))
    wl.append(float(list(data[data["School"] == i + " NCAA"]["W-L%"])[0]))
    sos.append(float(list(data[data["School"] == i + " NCAA"]["SOS"])[0]))

final["Overall Wins"] = w
final["W/L%"] = wl
final["SOS"] = sos

Akron
Alabama
Arizona
Auburn
Baylor
Boise State
BYU
Charleston
Clemson
Colgate
Colorado
Colorado State
Creighton
Dayton
Drake
Duke
Duquesne
Florida
Florida Atlantic
Gonzaga
Grambling State
Grand Canyon
Houston
Howard
Illinois
Iowa State
James Madison
Kansas
Kentucky
Long Beach State
Longwood
Marquette
McNeese
Michigan State
Mississippi State
Montana State
Morehead State
Nebraska
Nevada
New Mexico
North Carolina
North Carolina State
Northwestern
Oakland
Oregon
Purdue
Saint Mary's
Saint Peter's
Samford
San Diego State
South Carolina
South Dakota State
Stetson
TCU
Tennessee
Texas
Texas A&M
Texas Tech
UAB
UConn
Utah State
Vermont
Virginia
Wagner
Washington State
Western Kentucky
Wisconsin
Yale


In [4]:
final.to_excel("2024 fixed data.xlsx")

In [6]:
final

Team  Overall Wins   W/L%    SOS
0              Akron            24  0.706  -2.52
1            Alabama            21  0.656  11.39
2            Arizona            25  0.758   9.66
3             Auburn            27  0.794   8.04
4             Baylor            23  0.697  11.28
..               ...           ...    ...    ...
63            Wagner            17  0.531  -9.54
64  Washington State            24  0.727   6.89
65  Western Kentucky            22  0.667  -3.44
66         Wisconsin            22  0.629  11.62
67              Yale            22  0.710  -0.65

[68 rows x 4 columns]

Game Pairs:

Grambling vs Purdue: 0.36% (0.0036)

TCU vs Utah State: 16.11% (0.1756)

Gonzaga vs McNeese State: 24.73% (1.3971)

Kansas vs Samford: 21.28% (1.5474)

Oregon vs South Carolina: 63.71% (0.4509)

Akron vs Creighton: 50.26% (0.6985)

Colorado State vs Texas: 91.22% (2.4325)

Saint Peter's vs Tennessee: 1.87% (0.0189)



Creighton vs Oregon: 11.62% (2.1526)

Gonzaga vs Kansas: 29.69% (1.2142)

Tennessee vs Texas: 71.97% (0.3289)

Purdue vs Utah State: 79.16% (0.2337)

Gonzaga vs Purdue: 2.70% (0.0273)

Creighton vs Tennessee: 37.79% (0.4746)

Purdue vs Tennessee: 93.50% (0.0673)

In [32]:
# "2024 fixed data.xlsx"

final = pd.read_excel("2024 fixed data.xlsx")
games = pd.DataFrame(columns=["Overall Wins", "W/L%", "SOS"])

games.loc[len(games)] = np.asarray(final[final["Team"] == "Grambling State"][["Overall Wins", "W/L%", "SOS"]])[0] - np.asarray(final[final["Team"] == "Purdue"][["Overall Wins", "W/L%", "SOS"]])[0]
games.loc[len(games)] = np.asarray(final[final["Team"] == "TCU"][["Overall Wins", "W/L%", "SOS"]])[0] - np.asarray(final[final["Team"] == "Utah State"][["Overall Wins", "W/L%", "SOS"]])[0]
games.loc[len(games)] = np.asarray(final[final["Team"] == "Gonzaga"][["Overall Wins", "W/L%", "SOS"]])[0] - np.asarray(final[final["Team"] == "McNeese"][["Overall Wins", "W/L%", "SOS"]])[0]
games.loc[len(games)] = np.asarray(final[final["Team"] == "Kansas"][["Overall Wins", "W/L%", "SOS"]])[0] - np.asarray(final[final["Team"] == "Samford"][["Overall Wins", "W/L%", "SOS"]])[0]
games.loc[len(games)] = np.asarray(final[final["Team"] == "Oregon"][["Overall Wins", "W/L%", "SOS"]])[0] - np.asarray(final[final["Team"] == "South Carolina"][["Overall Wins", "W/L%", "SOS"]])[0]
games.loc[len(games)] = np.asarray(final[final["Team"] == "Akron"][["Overall Wins", "W/L%", "SOS"]])[0] - np.asarray(final[final["Team"] == "Creighton"][["Overall Wins", "W/L%", "SOS"]])[0]
games.loc[len(games)] = np.asarray(final[final["Team"] == "Colorado"][["Overall Wins", "W/L%", "SOS"]])[0] - np.asarray(final[final["Team"] == "Texas"][["Overall Wins", "W/L%", "SOS"]])[0]
games.loc[len(games)] = np.asarray(final[final["Team"] == "Saint Peter's"][["Overall Wins", "W/L%", "SOS"]])[0] - np.asarray(final[final["Team"] == "Tennessee"][["Overall Wins", "W/L%", "SOS"]])[0]

games.loc[len(games)] = np.asarray(final[final["Team"] == "Purdue"][["Overall Wins", "W/L%", "SOS"]])[0] - np.asarray(final[final["Team"] == "Utah State"][["Overall Wins", "W/L%", "SOS"]])[0]
games.loc[len(games)] = np.asarray(final[final["Team"] == "Gonzaga"][["Overall Wins", "W/L%", "SOS"]])[0] - np.asarray(final[final["Team"] == "Kansas"][["Overall Wins", "W/L%", "SOS"]])[0]
games.loc[len(games)] = np.asarray(final[final["Team"] == "Creighton"][["Overall Wins", "W/L%", "SOS"]])[0] - np.asarray(final[final["Team"] == "Oregon"][["Overall Wins", "W/L%", "SOS"]])[0]
games.loc[len(games)] = np.asarray(final[final["Team"] == "Tennessee"][["Overall Wins", "W/L%", "SOS"]])[0] - np.asarray(final[final["Team"] == "Texas"][["Overall Wins", "W/L%", "SOS"]])[0]

games.loc[len(games)] = np.asarray(final[final["Team"] == "Gonzaga"][["Overall Wins", "W/L%", "SOS"]])[0] - np.asarray(final[final["Team"] == "Purdue"][["Overall Wins", "W/L%", "SOS"]])[0]
games.loc[len(games)] = np.asarray(final[final["Team"] == "Creighton"][["Overall Wins", "W/L%", "SOS"]])[0] - np.asarray(final[final["Team"] == "Tennessee"][["Overall Wins", "W/L%", "SOS"]])[0]

games.loc[len(games)] = np.asarray(final[final["Team"] == "Purdue"][["Overall Wins", "W/L%", "SOS"]])[0] - np.asarray(final[final["Team"] == "Tennessee"][["Overall Wins", "W/L%", "SOS"]])[0]

games["Seed"] = [15, 1, -7, -9, 5, 11, 3, 13, -7, 1, -8, -5, 4, 1, 1]

games.to_excel("midwest.xlsx")

In [2]:
kaggle = pd.read_csv("kaggle.csv")

zo24 = pd.read_excel('final 2024.xlsx')

mn = {}
std = {}

cols = list(zo24.columns)[3:]

for k in cols:
    mn[k] = 0
    for i in range(len(zo24)):
        for j in range(i+1, len(zo24)):
            mn[k] += float(zo24.loc[i][k]) - float(zo24.loc[j][k])
    mn[k] /= len(zo24) * (len(zo24)-1)/2

for k in cols:
    std[k] = 0
    for i in range(len(zo24)):
        for j in range(i+1, len(zo24)):
            std[k] += (float(zo24.loc[i][k]) - float(zo24.loc[j][k]) - mn[k])**2
    std[k] = np.sqrt(std[k]/(len(zo24) * (len(zo24)-1)/2))

for k in cols:
    zo24[k] = (zo24[k] - mn[k])/std[k]

seed = float(list(zo24[zo24["Team"] == "Creighton"]["Seed"])[0]) - float(list(zo24[zo24["Team"] == "Oregon"]["Seed"])[0])
wl = float(list(zo24[zo24["Team"] == "Creighton"]["W/L%"])[0]) - float(list(zo24[zo24["Team"] == "Oregon"]["W/L%"])[0])
sos = float(list(zo24[zo24["Team"] == "Creighton"]["SOS"])[0]) - float(list(zo24[zo24["Team"] == "Oregon"]["SOS"])[0])
wd = float(list(zo24[zo24["Team"] == "Creighton"]["Overall Wins"])[0]) - float(list(zo24[zo24["Team"] == "Oregon"]["Overall Wins"])[0])

y = 1.28 * seed - 1.77 * wl + 2.37 * sos + 3.83 * wd
z = np.exp(y)/(1+np.exp(y))
z

0.1124303405294739

In [4]:
zo24

Team  Kaggle ID   Region      Seed  Overall Wins      W/L%  \
0              Akron       1103  MIDWEST  2.259371      5.245200  5.202875   
1            Alabama       1104     WEST  0.738255      4.577570  4.828067   
2            Arizona       1112     WEST  0.434032      5.467743  5.592676   
3             Auburn       1120     EAST  0.738255      5.912829  5.862538   
4             Baylor       1124     WEST  0.586144      5.022657  5.135410   
..               ...        ...      ...       ...           ...       ...   
63            Wagner       1447     WEST  2.563595      3.687398  3.891046   
64  Washington State       1450     EAST  1.194590      5.245200  5.360294   
65  Western Kentucky       1443    SOUTH  2.411483      4.800114  4.910525   
66         Wisconsin       1458    SOUTH  0.890367      4.800114  4.625670   
67              Yale       1463     EAST  2.107260      4.800114  5.232860   

         SOS  
0  -0.367200  
1   1.215000  
2   1.018221  
3   0.833953  
4   1.202488  
..       ...  
63 -1.165693  
64  0.703146  
65 -0.471846  
66  1.241162  
67 -0.154496  

[68 rows x 7 columns]

In [6]:
seed

-1.2168927560843688

In [7]:
wl

0.3223351027763446

In [8]:
sos

0.026161471079650944

In [9]:
wd

0.0

In [11]:
kaggle = pd.read_csv("kaggle.csv")

zo24 = pd.read_excel('final 2024.xlsx')

mn = {}
std = {}

cols = list(zo24.columns)[3:]

for k in cols:
    mn[k] = 0
    for i in range(len(zo24)):
        for j in range(i+1, len(zo24)):
            mn[k] += float(zo24.loc[i][k]) - float(zo24.loc[j][k])
    mn[k] /= len(zo24) * (len(zo24)-1)/2

for k in cols:
    std[k] = 0
    for i in range(len(zo24)):
        for j in range(i+1, len(zo24)):
            std[k] += (float(zo24.loc[i][k]) - float(zo24.loc[j][k]) - mn[k])**2
    std[k] = np.sqrt(std[k]/(len(zo24) * (len(zo24)-1)/2))

for k in cols:
    zo24[k] = (zo24[k] - mn[k])/std[k]

seed = float(list(zo24[zo24["Team"] == "Colorado State"]["Seed"])[0]) - float(list(zo24[zo24["Team"] == "Texas"]["Seed"])[0])
wl = float(list(zo24[zo24["Team"] == "Colorado State"]["W/L%"])[0]) - float(list(zo24[zo24["Team"] == "Texas"]["W/L%"])[0])
sos = float(list(zo24[zo24["Team"] == "Colorado State"]["SOS"])[0]) - float(list(zo24[zo24["Team"] == "Texas"]["SOS"])[0])
wd = float(list(zo24[zo24["Team"] == "Colorado State"]["Overall Wins"])[0]) - float(list(zo24[zo24["Team"] == "Texas"]["Overall Wins"])[0])

y = 1.28 * seed - 1.77 * wl + 2.37 * sos + 3.83 * wd
z = np.exp(y)/(1+np.exp(y))
z

0.9610840853903442

In [12]:
print(seed)
print(wl)
print(sos)
print(wd)

0.45633478353163826
0.667158701095226
-0.19336739493655186
1.112715854850193


In [258]:
zo24[zo24["Team"] == "UConn"]

Team  Kaggle ID Region      Seed  Overall Wins      W/L%       SOS
59  UConn       1163   EAST  0.281921      4.931264  4.980066  0.315203

In [256]:
zo24[zo24["Team"] == "Stetson"]

Team  Kaggle ID Region      Seed  Overall Wins      W/L%       SOS
52  Stetson       1391   EAST  2.563595      4.710338  4.760094 -0.457124

In [ ]:
kaggle = pd.read_csv("kaggle.csv")

zo24 = pd.read_excel('final 2024.xlsx')
zo223 = zo223[zo223["School"].str.contains("NCAA")].reset_index(drop=True)

zo323 = pd.read_excel('final 2023.xlsx')
seed = []
for i in range(len(zo223)):
    if len(list(zo323[zo323["School"] == zo223.loc[i]["School"][:-5]]["Seed 1"])) > 0:
        seed.append(int(list(zo323[zo323["School"] == zo223.loc[i]["School"][:-5]]["Seed 1"])[0]))
    else:
        seed.append(int(list(zo323[zo323["School.1"] == zo223.loc[i]["School"][:-5]]["Seed 2"])[0]))
zo223["Seed"] = seed

kids = []
for i in range(len(zo223)):
    kids.append(int(list(kaggle[kaggle["team name"] == (zo223.loc[i]["School"][:-5])]["TeamID"])[0]))
zo223["Kaggle ID"] = kids

kzo223 = zo223.sort_values("Kaggle ID").reset_index(drop=True)

mn = {}
std = {}

cols = list(zo223.columns)[1:]

for k in cols:
    mn[k] = 0
    for i in range(len(kzo223)):
        for j in range(i+1, len(kzo223)):
            mn[k] += float(kzo223.loc[i][k]) - float(kzo223.loc[j][k])
    mn[k] /= len(zo223) * (len(zo223)-1)/2

for k in cols:
    std[k] = 0
    for i in range(len(kzo223)):
        for j in range(i+1, len(kzo223)):
            std[k] += (float(kzo223.loc[i][k]) - float(kzo223.loc[j][k]) - mn[k])**2
    std[k] = np.sqrt(std[k]/(len(zo223) * (len(zo223)-1)/2))

for k in cols:
    zo223[k] = (zo223[k] - mn[k])/std[k]

a = {}
b = {}
diff = {}

cols = list(zo223.columns)[1:]

for j in cols:
    a[j] = []
    b[j] = []
    diff[j] = []

    for i in range(len(zo123)):
        
        val1 = float(list(zo223[zo223["School"] == str(zo123.loc[i]["School"] + " NCAA")][j])[0])
        val2 = float(list(zo223[zo223["School"] == str(zo123.loc[i]["School.1"] + " NCAA")][j])[0])

        a[j].append(val1)
        b[j].append(val2)
        diff[j].append(val1-val2)

for i in cols:
    zo123[str(i) + " Diff"] = diff[i]
for i in cols:
    zo123[str(i) + " 1"] = a[i]
for i in cols:
    zo123[str(i) + " 2"] = b[i]

zo123["Kaggle ID 1"] = zo123["Kaggle ID 1"].astype(int)
zo123["Kaggle ID 2"] = zo123["Kaggle ID 2"].astype(int)

zo123.to_excel("modified 2023.xlsx")

Reading Data

2023

In [205]:
kaggle = pd.read_csv("kaggle.csv")

zo = pd.ExcelFile('2023 all data.xlsx')
zo123 = pd.read_excel(zo, '2023 Training Data')
zo223 = pd.read_excel(zo, 'all 2023 tournament teams stats')
zo223 = zo223[zo223["School"].str.contains("NCAA")].reset_index(drop=True)

zo323 = pd.read_excel('final 2023.xlsx')
seed = []
for i in range(len(zo223)):
    if len(list(zo323[zo323["School"] == zo223.loc[i]["School"][:-5]]["Seed 1"])) > 0:
        seed.append(int(list(zo323[zo323["School"] == zo223.loc[i]["School"][:-5]]["Seed 1"])[0]))
    else:
        seed.append(int(list(zo323[zo323["School.1"] == zo223.loc[i]["School"][:-5]]["Seed 2"])[0]))
zo223["Seed"] = seed

kids = []
for i in range(len(zo223)):
    kids.append(int(list(kaggle[kaggle["team name"] == (zo223.loc[i]["School"][:-5])]["TeamID"])[0]))
zo223["Kaggle ID"] = kids

kzo223 = zo223.sort_values("Kaggle ID").reset_index(drop=True)

mn = {}
std = {}

cols = list(zo223.columns)[1:]

for k in cols:
    mn[k] = 0
    for i in range(len(kzo223)):
        for j in range(i+1, len(kzo223)):
            mn[k] += float(kzo223.loc[i][k]) - float(kzo223.loc[j][k])
    mn[k] /= len(zo223) * (len(zo223)-1)/2

for k in cols:
    std[k] = 0
    for i in range(len(kzo223)):
        for j in range(i+1, len(kzo223)):
            std[k] += (float(kzo223.loc[i][k]) - float(kzo223.loc[j][k]) - mn[k])**2
    std[k] = np.sqrt(std[k]/(len(zo223) * (len(zo223)-1)/2))

for k in cols:
    zo223[k] = (zo223[k] - mn[k])/std[k]

a = {}
b = {}
diff = {}

cols = list(zo223.columns)[1:]

for j in cols:
    a[j] = []
    b[j] = []
    diff[j] = []

    for i in range(len(zo123)):
        
        val1 = float(list(zo223[zo223["School"] == str(zo123.loc[i]["School"] + " NCAA")][j])[0])
        val2 = float(list(zo223[zo223["School"] == str(zo123.loc[i]["School.1"] + " NCAA")][j])[0])

        a[j].append(val1)
        b[j].append(val2)
        diff[j].append(val1-val2)

for i in cols:
    zo123[str(i) + " Diff"] = diff[i]
for i in cols:
    zo123[str(i) + " 1"] = a[i]
for i in cols:
    zo123[str(i) + " 2"] = b[i]

zo123["Kaggle ID 1"] = zo123["Kaggle ID 1"].astype(int)
zo123["Kaggle ID 2"] = zo123["Kaggle ID 2"].astype(int)

zo123.to_excel("modified 2023.xlsx")

2022

In [199]:
kaggle = pd.read_csv("kaggle.csv")

zo = pd.ExcelFile('2022 all data.xlsx')
zo122 = pd.read_excel(zo, '2022 Training Data')
zo222 = pd.read_excel(zo, 'all 2022 tournament teams stats')

zo222 = zo222[zo222["School"].str.contains("NCAA")].reset_index(drop=True)

zo322 = pd.read_excel('final 2022.xlsx')
seed = []
for i in range(len(zo222)):
    if len(list(zo322[zo322["School"] == zo222.loc[i]["School"][:-5]]["Seed 1"])) > 0:
        seed.append(int(list(zo322[zo322["School"] == zo222.loc[i]["School"][:-5]]["Seed 1"])[0]))
    else:
        seed.append(int(list(zo322[zo322["School.1"] == zo222.loc[i]["School"][:-5]]["Seed 2"])[0]))
zo222["Seed"] = seed

kids = []
for i in range(len(zo222)):
    kids.append(int(list(kaggle[kaggle["team name"] == (zo222.loc[i]["School"][:-5])]["TeamID"])[0]))
zo222["Kaggle ID"] = kids

kzo222 = zo222.sort_values("Kaggle ID").reset_index(drop=True)

mn = {}
std = {}

cols = list(zo222.columns)[1:]

for k in cols:
    mn[k] = 0
    for i in range(len(kzo222)):
        for j in range(i+1, len(kzo222)):
            mn[k] += float(kzo222.loc[i][k]) - float(kzo222.loc[j][k])
    mn[k] /= len(zo222) * (len(zo222)-1)/2

for k in cols:
    std[k] = 0
    for i in range(len(kzo222)):
        for j in range(i+1, len(kzo222)):
            std[k] += (float(kzo222.loc[i][k]) - float(kzo222.loc[j][k]) - mn[k])**2
    std[k] = np.sqrt(std[k]/(len(zo222) * (len(zo222)-1)/2))

for k in cols:
    zo222[k] = (zo222[k] - mn[k])/std[k]

a = {}
b = {}
diff = {}

cols = list(zo222.columns)[1:]

for j in cols:
    a[j] = []
    b[j] = []
    diff[j] = []

    for i in range(len(zo122)):
        
        val1 = float(list(zo222[zo222["School"] == str(zo122.loc[i]["School"] + " NCAA")][j])[0])
        val2 = float(list(zo222[zo222["School"] == str(zo122.loc[i]["School.1"] + " NCAA")][j])[0])

        a[j].append(val1)
        b[j].append(val2)
        diff[j].append(val1-val2)

for i in cols:
    zo122[str(i) + " Diff"] = diff[i]
for i in cols:
    zo122[str(i) + " 1"] = a[i]
for i in cols:
    zo122[str(i) + " 2"] = b[i]

zo122["Kaggle ID 1"] = zo122["Kaggle ID 1"].astype(int)
zo122["Kaggle ID 2"] = zo122["Kaggle ID 2"].astype(int)

zo122.to_excel("modified 2022.xlsx")

2019

In [248]:
kaggle = pd.read_csv("kaggle.csv")
zo119 = pd.read_excel("2019 Training Data.xlsx")
zo219 = pd.read_excel('all 2019 tournament stats.xlsx')
zo219 = zo219[zo219["School"].str.contains("NCAA")].reset_index(drop=True)

zo319 = pd.read_excel('final 2019.xlsx')
seed = []
for i in range(len(zo219)):
    if len(list(zo319[zo319["School"] == zo2.loc[i]["School"][:-5]]["Seed 1"])) > 0:
        seed.append(int(list(zo319[zo319["School"] == zo219.loc[i]["School"][:-5]]["Seed 1"])[0]))
    else:
        seed.append(int(list(zo319[zo319["School.1"] == zo219.loc[i]["School"][:-5]]["Seed 2"])[0]))
zo219["Seed"] = seed

kids = []
for i in range(len(zo219)):
    kids.append(int(list(kaggle[kaggle["team name"] == (zo219.loc[i]["School"][:-5])]["TeamID"])[0]))
zo219["Kaggle ID"] = kids

kzo219 = zo219.sort_values("Kaggle ID").reset_index(drop=True)

mn = {}
std = {}

cols = list(zo219.columns)[1:]

for k in cols:
    mn[k] = 0
    for i in range(len(kzo219)):
        for j in range(i+1, len(kzo219)):
            mn[k] += (kzo219.loc[i][k]).astype(Decimal) - (kzo219.loc[j][k]).astype(Decimal)
    mn[k] /= len(zo219) * (len(zo219)-1)/2

for k in cols:
    std[k] = 0
    for i in range(len(kzo219)):
        for j in range(i+1, len(kzo219)):
            std[k] += ((kzo219.loc[i][k]).astype(Decimal) - (kzo219.loc[j][k]).astype(Decimal) - mn[k])**2
    std[k] = np.sqrt(std[k]/(len(zo219) * (len(zo219)-1)/2))

for k in cols:
    zo219[k] = (zo219[k] - mn[k])/std[k]

a = {}
b = {}
diff = {}

cols = list(zo219.columns)[1:]

for j in cols:
    a[j] = []
    b[j] = []
    diff[j] = []

    for i in range(len(zo119)):
        
        val1 = (list(zo219[zo219["School"] == str(zo119.loc[i]["School"] + " NCAA")][j])[0])
        val2 = (list(zo219[zo219["School"] == str(zo119.loc[i]["School.1"] + " NCAA")][j])[0])

        a[j].append(val1)
        b[j].append(val2)
        diff[j].append(val1-val2)

for i in cols:
    zo119[str(i) + " Diff"] = diff[i]
for i in cols:
    zo119[str(i) + " 1"] = a[i]
for i in cols:
    zo119[str(i) + " 2"] = b[i]

zo119["Kaggle ID 1"] = zo119["Kaggle ID 1"].astype(int)
zo119["Kaggle ID 2"] = zo119["Kaggle ID 2"].astype(int)

zo119.to_excel("modified 2019.xlsx")

In [182]:
zo219.to_excel("Adjustment Testing/teamdata.xlsx")

In [240]:
zo119.columns

Index(['Date', 'School', 'School.1', 'Result', 'G Diff', 'Overall W Diff',
       'Overall L Diff', 'W-L% Diff', 'SRS Diff', 'SOS Diff',
       'Conference W Diff', 'Conference L Diff', 'Home W Diff', 'Home L Diff',
       'Away W Diff', 'Away L Diff', 'Tm. Diff', 'Opp. Diff', 'Pace Diff',
       'ORtg Diff', 'FTr Diff', '3PAr Diff', 'TS% Diff', 'TRB% Diff',
       'AST% Diff', 'STL% Diff', 'BLK% Diff', 'eFG% Diff', 'TOV% Diff',
       'ORB% Diff', 'FT/FGA Diff', 'Seed Diff', 'Kaggle ID Diff', 'G 1',
       'Overall W 1', 'Overall L 1', 'W-L% 1', 'SRS 1', 'SOS 1',
       'Conference W 1', 'Conference L 1', 'Home W 1', 'Home L 1', 'Away W 1',
       'Away L 1', 'Tm. 1', 'Opp. 1', 'Pace 1', 'ORtg 1', 'FTr 1', '3PAr 1',
       'TS% 1', 'TRB% 1', 'AST% 1', 'STL% 1', 'BLK% 1', 'eFG% 1', 'TOV% 1',
       'ORB% 1', 'FT/FGA 1', 'Seed 1', 'Kaggle ID 1', 'G 2', 'Overall W 2',
       'Overall L 2', 'W-L% 2', 'SRS 2', 'SOS 2', 'Conference W 2',
       'Conference L 2', 'Home W 2', 'Home L 2', 'Awa

In [246]:
test = zo122

for i in list(test.columns)[4:30]:
    test[i] = test[i]/np.std(test[i])

test[["SOS Diff"]]

SOS Diff
0  -1.728961
1   0.644316
2   0.052052
3  -0.467059
4   1.681130
..       ...
62  1.346311
63 -0.250411
64 -0.230716
65  0.336226
66 -0.402346

[67 rows x 1 columns]

In [ ]:
seed = (final2023["Seed Diff"] - np.mean(final2023["Seed Diff"]))/(np.std(final2023["Seed Diff"]))
sos = (final2023["SOS Diff"]-np.mean(final2023["SOS Diff"]))/(np.std(final2023["SOS Diff"]))
wd = (final2023["Overall W Diff"]-np.mean(final2023["Overall W Diff"]))/(np.std(final2023["Overall W Diff"]))
wl = (final2023["W-L% Diff"] - np.mean(final2023["W-L% Diff"]))/(np.std(final2023["W-L% Diff"]))

seed2 = (final2019["Seed Dif"] - np.mean(final2019["Seed Dif"]))/(np.std(final2019["Seed Dif"]))
sos2 = (final2019["SOS Diff"]-np.mean(final2019["SOS Diff"]))/(np.std(final2019["SOS Diff"]))
wd2 = (final2019["Overall W Diff"]-np.mean(final2019["Overall W Diff"]))/(np.std(final2019["Overall W Diff"]))
wl2 = (final2019["W-L% Diff"] - np.mean(final2019["W-L% Diff"]))/(np.std(final2019["W-L% Diff"]))

seed2 = (final2019["Seed Dif"] - np.mean(final2019["Seed Dif"]))/(np.std(final2019["Seed Dif"]))
sos2 = (final2019["SOS Diff"]-np.mean(final2019["SOS Diff"]))/(np.std(final2019["SOS Diff"]))
wd2 = (final2019["Overall W Diff"]-np.mean(final2019["Overall W Diff"]))/(np.std(final2019["Overall W Diff"]))
wl2 = (final2019["W-L% Diff"] - np.mean(final2019["W-L% Diff"]))/(np.std(final2019["W-L% Diff"]))

1.057 seed diff - 1.70 w/l % diff + 2.64 strength of schedule diff + 3.30 overall w diff